internal/core/api.rs

```diff
@@ -536,6 +536,9 @@ pub trait ComponentHandle {
     where
         Self: Sized;
 
+    /// Return a Weak handle to a `ComponentVTable`
+    fn as_weak_component(&self) -> ComponentWeak;
```

Comment:
Don't add this function to this public trait.  There is already as_weak that serve this exact purpose.
Just add a crate local accessor in Weak to get the inner.

The reason is that adding this adds
 1. a public function
 2. something that every implementation needs to re-implement

Re-using as_weak is much simpler